tasks/preload.js

```diff
@@ -136,7 +136,14 @@ module.exports = function (grunt) {
 					iMinor = parseInt(aVersionMatch[2], 10);
 				}
 
-				if (options.compatVersion === "edge" || (iMajor === 1 && iMinor >= 40) || iMajor > 1) {
+				if (options.compatVersion === "edge" || (iMajor === 1 && iMinor >= 54) || iMajor > 1) {
+					// Build library-preload as .js file
+					preloadInfo.ext = ".js";
+					preloadInfo.processContent = function(content, preloadObject) {
+						// workaround for new modules based on sap.ui.predefine
```

Comment:
I would rather call this the new API, not a workaround. I also think it's not really coupled to the sap.ui.predefine format.